packages/circle-demo-webapp/app/lib/utils.ts

```diff
@@ -22,3 +22,78 @@ export async function callFetch<ReturnType extends object>(
   });
   return (await res.json()) as ReturnType;
 }
+
+export type ValidInputTypes = Uint8Array | bigint | string | number | boolean;
+export const isUint8Array = (data: ValidInputTypes): data is Uint8Array =>
+  data instanceof Uint8Array ||
+  data?.constructor?.name === 'Uint8Array' ||
+  data?.constructor?.name === 'Buffer';
+
+export function uint8ArrayToHexString(uint8Array: Uint8Array): string {
+  let hexString = '0x';
+  for (const e of uint8Array) {
+    const hex = e.toString(16);
+    hexString += hex.length === 1 ? `0${hex}` : hex;
+  }
+  return hexString;
+}
+
+export const isHexStrict = (hex: ValidInputTypes) =>
+  typeof hex === 'string' && /^((-)?0x[0-9a-f]+|(0x))$/i.test(hex);
+
+export const isAddress = (value: string): boolean => {
+  if (typeof value !== 'string' && !isUint8Array(value)) {
```

Comment:
why do we expect a Uint8Array here? I think the address from the form is always a string, no? this validation function could be simplified then.

Reply:
this is isAddress logic from web3js

Reply:
fixed